hal/src/nRF52840/ble_hal.cpp

```diff
@@ -327,7 +327,7 @@ class BleObject::Broadcaster {
     volatile hal_ble_auto_adv_cfg_t autoAdvCfg_;    /**< Automatic advertising configuration. */
     uint8_t advHandle_;                             /**< Advertising handle. */
     hal_ble_adv_params_t advParams_;                /**< Current advertising parameters. */
-    uint8_t advData_[BLE_MAX_ADV_DATA_LEN];         /**< Current advertising data. */
+    uint8_t advData_[BLE_MAX_ADV_DATA_LEN_EXT];     /**< Current advertising data. */
```

Comment:
We should use `BLE_MAX_ADV_DATA_LEN_EXT_CONNECTABLE` for the advertising data buffer.

Reply:
> We should use `BLE_MAX_ADV_DATA_LEN_EXT_CONNECTABLE` for the advertising data buffer.

What if a customer wants to make an application that advertises the full 255 bytes in non-connectable mode?